### PR TITLE
Redefine canCrabClimb to canCrazyCrabClimb

### PR DESCRIPTION
**Pull request by kjbranch**
- Move easier variants of `canCrabClimb` to `canTrickyUseFrozenEnemies`
- Move harder variants to `canCrazyCrabClimb`
- Reevaluate what is notable and not, and add some `reusableRoomwideNotable`

Note that this can not be merged until after #645. It will need to be merged into this and then add some minor edits.

### CODE REVIEW

File: region/maridia/outer.json [hunk 1]
```diff
@@ -3125,10 +3148,10 @@
                 },
                 {
                   "name": "Mt. Everest Right Bottom Half Crab Climb with Supers and HiJump",
-                  "notable": true,
+                  "notable": false,
```
Comment (osse101) [inline region/maridia/outer.json:3151]: These still have a `reusableRoomwideNotable`

File: region/maridia/outer.json [hunk 2]
```diff
@@ -889,6 +889,24 @@
           "homeNodes": [ 5, 7 ]
         }
       ],
+      "reusableRoomwideNotable": [
+        {
+          "name": "Main Street Crab Climb with Only Ice and Supers",
```
Comment (osse101) [inline region/maridia/outer.json:894]: This isn't reused, not that it _needs_ to be.

Comment (kjbranch) [reply region/maridia/outer.json:894]: Yeah, the second half (mid climb with supers) is in #645

File: region/maridia/outer.json [hunk 3]
```diff
@@ -5256,7 +5258,7 @@
                       "type": "Super",
                       "count": 1
                     }},
-                    "canCrabClimb"
+                    "canTrickyUseFrozenEnemies"
```
Comment (osse101) [inline region/maridia/outer.json:5261]: I'd update this to notable.

Comment (osse101) [reply region/maridia/outer.json:5261]: Needs a rename now too.  Crab Hole Crab Climb No Jump Assist.

**Comment by osse101** [comment]
👍

**Pull request merged**